File: englishmate/src/types/ListeningLessonDetailDTO.ts
export interface TranscriptLine {
  text: string;
  start: number;
}

export interface ListeningLessonDetailDTO {
  id: number;
  title: string;
  imageUrl: string;
  description: string;
  audioUrl: string;
  duration: number;
  transcript: TranscriptLine[];
  level: string;
  rating: number;
  exercise: string;
  completedCount: number;
  type: string;
  lessonTitle: string;
  courseTitle: string;
}

File: englismate_api/core_service/src/main/java/com/englishmate/core_service/dto/lesson/ListeningLessonDetailDTO.java
package com.englishmate.core_service.dto.lesson;

import com.englishmate.core_service.converter.dto.QuizQuestion;
import com.englishmate.core_service.converter.dto.Transcript;
import com.englishmate.core_service.entity.enumerations.LessonLevel;
import com.englishmate.core_service.entity.enumerations.QuestionType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ListeningLessonDetailDTO {
    private Long id;
    private String title;
    private String imageUrl;
    private String description;
    private String audioUrl;
    private Integer duration;
    private List<Transcript> transcript;
    private String level;
    private Integer rating;
    private List<QuizQuestion> exercise;
    private Long completedCount;
    private QuestionType type;
    private String lessonTitle;
    private String courseTitle;
} 
